src/Models/HydrostaticFreeSurfaceModels/SplitExplicitFreeSurfaces/split_explicit_free_surface.jl

```diff
@@ -202,16 +204,50 @@ function materialize_free_surface(free_surface::SplitExplicitFreeSurface, veloci
                                     timestepper)
 end
 
-# (p = 2, q = 4, r = 0.18927) minimize dispersion error from Shchepetkin and McWilliams (2005): https://doi.org/10.1016/j.ocemod.2004.08.002
+""" An internal function that stores the averaging kernel 
+and the last fractional time stepped in the subcycling """
+struct AveragingKernel{FT}
+    kernel :: Function
```

Comment:
This isn't concretely typed because `Function` is an abstract type. You need to use a type parameter.

(Also as a passing comment, it's usually not a good idea to subtype or dispatch on `::Function`, because it is not very specific. Callable objects are not functions, and `Function` is only an abstract type).